### PR TITLE
66e60f

### PR DESCRIPTION
**Pull request by eskila**
Protocol code is finished

The protocol requester wanted to use an m20 to distribute the water, then p20 and p300 to distribute samples.
I thought it would make more sense to just distribute the water with the p300, rather than having 2 protocols and
a pipette change.

See protocol description here: https://docs.google.com/document/d/1kemf6lV1ISjTAPRMmQii7PzVsjCxV0RUBHqc0h-8lEg/edit?usp=sharing

**Edit 12/20/21: Everything is finished, but I can't make the protobuilds because of the following error**

`

['mass_spec_sample_prep.ot2.apiv2.py']  
[]
[]
Traceback (most recent call last):
  File "protolib/traverse_errors.py", line 120, in <module>
    write_metadata_to_file(PROTOCOL_DIR)
  File "protolib/traverse_errors.py", line 115, in write_metadata_to_file
    'status': get_status(metadata)},
  File "protolib/traverse_errors.py", line 86, in get_status
    errors = get_errors(file_data['files'])
  File "protolib/traverse_errors.py", line 67, in get_errors
    raise ValueError('Found 0 protocol files required at least 1')
ValueError: Found 0 protocol files required at least 1
make: *** [parse-errors] Error 1`

### CODE REVIEW

File: protocols/66e60f/66e60f.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        volume = float(line[3])
+
+        ctx.comment("Normalizing sample {} with concentration {}"
+                    .format(description, concentration))
+
+        if volume <= 20:
+            p20.pick_up_tip()
+            p20.transfer(volume,
+                         target_plate.wells_by_name()[well],
+                         water_well, new_tip="never")
```
Comment (eskila) [inline protocols/66e60f/66e60f.py:284]: Dispense into waste well instead

File: protocols/66e60f/66e60f.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    #              [0]       [1]           [2]                [3]
+    data = [
+        [val.strip().upper() for val in line.split(',')
+            if val != '']
+        for line in input_csv.splitlines()[1:]
+        if line and line.split(',')[0]]
+
+    # Convert any well designation in column 1 from [A-H]0[1-9] to [A-H][1-9]
+    # e.g. A01 -> A1 etc.
+    pattern = re.compile('[A-H]0[1-9]')
```
Comment (eskila) [inline protocols/66e60f/66e60f.py:53]: Could be less computationally intensive, e.g. check if the second character is a 0
Nick: return well[0] + str(int(well[1:]))

File: protocols/66e60f/66e60f.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    for line in data:
+        well = line[0]
+        description = line[1]
+        concentration = line[2]
+        volume = float(line[3])
+
+        ctx.comment("Normalizing sample {} with concentration {}"
+                    .format(description, concentration))
+
+        if volume <= 20:
```
Comment (eskila) [inline protocols/66e60f/66e60f.py:280]: Could be a function or an if statement
if vol <= 20: pip = p20, else: pip = p300

Comment (eskila) [reply protocols/66e60f/66e60f.py:280]: pip = p20 if vol <= 20 else p300

**Comment by chazmanc** [comment]
This protocol was added #3115

**Pull request closed without merging**